src/main/java/io/cdap/plugin/date/DateTransformConfig.java

```diff
@@ -0,0 +1,290 @@
+      stringList.add(field.trim());
+    }
+    return stringList;
+  }
+
+  public List<String> getTargetFields() {
+    String[] sourceFields = targetFields.split(",");
+    List<String> stringList = new ArrayList<>();
+    for (String field : sourceFields) {
+      stringList.add(field.trim());
+    }
+    return stringList;
+  }
+
+  public String getTargetFieldsString() {
+    return targetFields;
+  }
+
+  public String getSourceFieldsString() {
+    return sourceFields;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    try {
+      Schema outputSchema = getSchema();
+      if (!containsMacro(PROPERTY_TARGET_FIELDS)) {
+        List<String> targetFields = this.getTargetFields();
+        for (String targetField : targetFields) {
+          if (outputSchema.getField(targetField) == null) {
+            failureCollector.addFailure(String.format("Target field '%s' is not present in output schema",
+                                                      targetField), null)
+              .withConfigElement(PROPERTY_TARGET_FIELDS, targetField)
+              .withOutputSchemaField(targetField);
+          }
+        }
+      }
+    } catch (IllegalArgumentException e) {
+      failureCollector.addFailure("Output schema cannot be parsed.", null)
+        .withStacktrace(e.getStackTrace());
+    }
+
+    validateDateFormat(failureCollector, targetFormat, PROPERTY_TARGET_FORMAT);
+    validateDateFormat(failureCollector, sourceFormat, PROPERTY_SOURCE_FORMAT);
+
+    if (!containsMacro(PROPERTY_TARGET_FIELDS) && !containsMacro(PROPERTY_SOURCE_FIELDS)) {
+      if (this.getTargetFields().size() != this.getSourceFields().size()) {
+        failureCollector.addFailure("Target and source fields must contain the same number of fields.", null)
+          .withConfigProperty(PROPERTY_SOURCE_FIELDS)
+          .withConfigProperty(PROPERTY_TARGET_FIELDS);
+      }
+    }
+
+    if (inputSchema == null) {
+      failureCollector.addFailure("Input schema cannot be null.", null);
+      throw failureCollector.getOrThrowException();
+    }
+
+    List<String> sourceFields = getSourceFields();
+    for (String field : sourceFields) {
+      Schema.Field inputField = inputSchema.getField(field);
+      if (inputField == null) {
+        failureCollector.addFailure(String.format("Source field '%s' is not present in input schema.",
+                                                  field), null)
+          .withConfigProperty(PROPERTY_SOURCE_FIELDS)
+          .withInputSchemaField(field);
+      } else {
+        Schema inputFieldSchema = inputField.getSchema();
+
+        if (inputFieldSchema.isNullable()) {
+          inputFieldSchema = inputFieldSchema.getNonNullable();
+        }
+
+        if (inputFieldSchema.getType() != Schema.Type.STRING && inputFieldSchema.getType() != Schema.Type.LONG) {
+          failureCollector.addFailure(String.format("Source field '%s' must be of type 'string' or 'long'" +
```

Comment:
Error message: `Source Field '%s' is unexpected type '%s'.`
corrective action: `Supported types are 'string' or 'long'.`

Reply:
Done